### PR TITLE
Update the ACL package to use git-checkout instead of fetch

### PR DESCRIPTION
**Pull request by dakaneye**
Fixes:  Updates ACL to use git-checkout vs fetch, as a means to bias away fetching compiled tars, which is related the xz backdoor exploit earlier this year

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [ ] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [ ] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

**Review by rawlingsj — DISMISSED** [review]

**Comment by rawlingsj** [comment]
You will probably need to use this at the top https://github.com/wolfi-dev/os/blob/c5248923a7545452ee1515eeb137edf4a626ca41/gitlab-exporter.yaml#L4

This is to suppress the Wolfi Lint error as the git provider is not github

File: acl.yaml [hunk 1]
```diff
@@ -1,25 +1,33 @@
...
   contents:
     packages:
+      - autoconf
+      - automake
       - attr-dev
       - build-base
       - busybox
       - ca-certificates-bundle
+      - libtool
+      - gettext-dev
```
Comment (rawlingsj) [inline acl.yaml:19]: I don't know, but wonder if this is causing the build failure, are both these needed now?

Comment (dakaneye) [reply acl.yaml:19]: Now that we have to call the `autogen.sh` script, we require a tool called `autopoint` which is included in gettext-dev, and needed `aclocal` which is in `automake`. The other make commands needed additional context too when I was running `make package/acl`, specifically libtool.

**Comment by xnox** [comment]
> You will probably need to use this at the top https://github.com/wolfi-dev/os/blob/c5248923a7545452ee1515eeb137edf4a626ca41/gitlab-exporter.yaml#L4
> 
> This is to suppress the Wolfi Lint error as the git provider is not github

That should be out of date now.

We fixed wolfi lint to only raise alarm if the URL is GitHub. Non GitHub URLs no longer trigger the lint check "must use GitHub" for updates.

Actually should mass drop that lint comment.

**Comment by dakaneye** [comment]
Regarding the build failure: 
```2024/06/05 18:35:25 ERRO unresolvable cycle: libarchive:3.7.4-r0@local -> acl-dev:2.3.2-r2@local, caused by: acl-dev:2.3.2-r2@local -> acl:2.3.2-r2@local -> git:2.45.2-r0@local -> curl-dev:8.8.0-r0@local -> curl:8.8.0-r0@local -> brotli-dev:1.1.0-r2@local -> brotli:1.1.0-r2@local -> cmake:3.29.0-r0@local -> libarchive-dev:3.7.4-r0@local -> libarchive:3.7.4-r0@local```

I'm not sure what is causing this, I didn't introduce a git dependency into the build environment?

**Comment by dakaneye** [comment]
> Regarding the build failure: `2024/06/05 18:35:25 ERRO unresolvable cycle: libarchive:3.7.4-r0@local -> acl-dev:2.3.2-r2@local, caused by: acl-dev:2.3.2-r2@local -> acl:2.3.2-r2@local -> git:2.45.2-r0@local -> curl-dev:8.8.0-r0@local -> curl:8.8.0-r0@local -> brotli-dev:1.1.0-r2@local -> brotli:1.1.0-r2@local -> cmake:3.29.0-r0@local -> libarchive-dev:3.7.4-r0@local -> libarchive:3.7.4-r0@local`
> 
> I'm not sure what is causing this, I didn't introduce a git dependency into the build environment? Perhaps now that it's not "fetching" it needs git in the test?

Marking this as a draft, this won't work until @xnox 's git-bootstrap PR lands

**Review by pnasrat — APPROVED** [review]
LGTM

**Pull request merged**